packages/legend-studio/src/stores/sidebar-state/ConflictResolutionState.ts

```diff
@@ -66,22 +62,75 @@ export class ConflictResolutionState {
   mergeEditorStates: EntityChangeConflictEditorState[] = [];
 
   constructor(editorStore: EditorStore, sdlcState: EditorSDLCState) {
-    makeAutoObservable(this, {
+    this.editorStore = editorStore;
+    this.sdlcState = sdlcState;
+  }
+
+  abstract get resolutions(): EntityChangeConflictResolution[];
+
+  abstract openConflict(conflict: EntityChangeConflict): void;
+
+  abstract closeState(conflict: EntityChangeConflictEditorState): void;
+
+  abstract handleResolution(resolution: EntityChangeConflictResolution): void;
+
+  abstract onMarkAsResolved(
+    conflict: EntityChangeConflictEditorState,
+  ): GeneratorFn<void>;
+
+  removeMergeEditorState(
+    mergeEditorState: EntityChangeConflictEditorState,
+  ): void {
+    deleteEntry(this.mergeEditorStates, mergeEditorState);
+  }
+}
+
+export class ConflictResolutionState extends AbstractConflictResolutionState {
```

Comment:
Do me a favor and rename + move this out to a separate file called `WorkspaceUpdateConflictResolutionState`